lightning/src/onion_message/messenger.rs

```diff
@@ -386,8 +386,8 @@ where
 			let node_announcement = network_graph
 				.node(&NodeId::from_pubkey(&first_node))
 				.and_then(|node_info| node_info.announcement_info.as_ref())
-				.and_then(|announcement_info| announcement_info.announcement_message.as_ref())
-				.map(|node_announcement| &node_announcement.contents);
+				.and_then(|announcement_info| announcement_info.announcement_message().cloned())
```

Comment:
This will fix it:

```diff
diff --git a/lightning/src/onion_message/messenger.rs b/lightning/src/onion_message/messenger.rs
index 283186bce..1de5d5553 100644
--- a/lightning/src/onion_message/messenger.rs
+++ b/lightning/src/onion_message/messenger.rs
@@ -386,8 +386,9 @@ where
                        let node_announcement = network_graph
                                .node(&NodeId::from_pubkey(&first_node))
                                .and_then(|node_info| node_info.announcement_info.as_ref())
-                               .and_then(|announcement_info| announcement_info.announcement_message().cloned())
-                               .map(|node_announcement| node_announcement.contents);
+                               .and_then(|announcement_info| announcement_info.announcement_message()
+                                       .map(|node_announcement| &node_announcement.contents)
+                               );
 
                        match node_announcement {
                                Some(node_announcement) if node_announcement.features.supports_onion_messages() => {
```